docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -8,7 +8,28 @@
 
 Proposed
 
-## Context
+## Abstract
+
+Fully deterministic structure serializaiton, which works across many languages / clients,
+is needed for structure signature use-case. We need to be sure that whenever we serialize
+and deserialize a data structure, no matter in which supported language, the raw bytes
+will stay the same. Protobufs are not deterministic.
+
+The major use-case for deterministic protobuf serialization is:
+
+* Digital Signatures
+* Merkle Tree
+
+Currently, for Block signature we are using a workaround: we create a new [TxRaw](https://github.com/cosmos/cosmos-sdk/blob/9e85e81e0e8140067dd893421290c191529c148c/proto/cosmos/tx/v1beta1/tx.proto#L30)
+instance (as explained in[adr-020-protobuf-transaction-encoding](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-020-protobuf-transaction-encoding.md#transactions))
+by converting all Tx fields to bytes on the client side. This adds additional manual
+step when sending and signing transactions.
+
+Here we propose how to limit the protobufs in order to assure
+deterministic serializaiton.
```

Comment:
Same thing, in "Context", there's this sentence:

> This document describes a deterministic serialization scheme for a subset of protobuf documents, that covers this use case but can be reused in other cases as well.

Reply:
Thanks for catching. I've merged this sections.